packages/loot-core/src/shared/schedules.ts

```diff
@@ -369,6 +369,24 @@ export function getUpcomingDays(upcomingLength = '7'): number {
       );
     }
     default:
+      if (upcomingLength.includes('-')) {
+        const [num, unit] = upcomingLength.split('-');
+        const value = Math.max(1, parseInt(num, 10));
+        switch (unit) {
+          case 'day':
+            return value;
+          case 'week':
+            return value * 7;
+          case 'month':
+            const future = monthUtils.addMonths(today, value);
+            return monthUtils.differenceInCalendarDays(future, month);
+          case 'year':
+            const futureYear = monthUtils.addYears(today, value);
+            return monthUtils.differenceInCalendarDays(futureYear, month);
```

Comment:
It's coming up one day short of expected now, adding one to both will do the trick. That's what I did with oneMonth

<img width="1176" alt="image" src="https://github.com/user-attachments/assets/72ebe013-2896-4e73-8d25-f34877b0da40" />

```suggestion
          case 'month':
            const future = monthUtils.addMonths(today, value);
            return monthUtils.differenceInCalendarDays(future, month) + 1;
          case 'year':
            const futureYear = monthUtils.addYears(today, value);
            return monthUtils.differenceInCalendarDays(futureYear, month) + 1;
```